packages/hooks/use-focus-controller/index.ts

```diff
@@ -54,12 +54,13 @@ export function useFocusController<T extends { focus: () => void }>(
     afterBlur?.()
   }
 
-  const handleClick = () => {
-    if (
-      wrapperRef.value?.contains(document.activeElement) &&
-      wrapperRef.value !== document.activeElement
-    )
-      return
+  const handleClick = (event: MouseEvent) => {
+    const shouldSkipFocus =
+      (event.target as Element)?.closest('.clear-icon') ||
+      (wrapperRef.value?.contains(document.activeElement) &&
+        wrapperRef.value !== document.activeElement)
+
+    if (shouldSkipFocus) return
```

Comment:
Maybe we can refer to `handleFocus` and add a `beforeClick` to handle special situations. 🤔